sdk/core/perf/src/base_test.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/perf/base_test.hpp"
+#include "azure/core/http/policies/policy.hpp"
+#include "azure/core/internal/http/pipeline.hpp"
+
+#include <functional>
+#include <string>
+#include <vector>
+
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core;
+namespace {
+
+class ProxyPolicy final : public HttpPolicy {
+private:
+  Azure::Perf::BaseTest* m_testContext;
+
+public:
+  ProxyPolicy(Azure::Perf::BaseTest* proxyManager) : m_testContext(proxyManager) {}
+
+  // copy
+  ProxyPolicy(ProxyPolicy const& other) : ProxyPolicy{other.m_testContext} {}
+
+  // move
+  ProxyPolicy(ProxyPolicy&& other) : m_testContext{other.m_testContext} {}
+
+  std::unique_ptr<RawResponse> Send(
+      Request& request,
+      NextHttpPolicy nextPolicy,
+      Context const& context) const override
+  {
+    std::string const recordId(m_testContext->m_recordId);
+    if (recordId.empty())
+    {
+      return nextPolicy.Send(request, context);
+    }
+
+    // Use a new request to redirect
+    auto redirectRequest = Azure::Core::Http::Request(
+        request.GetMethod(), Azure::Core::Url(m_testContext->m_proxy), request.GetBodyStream());
+    redirectRequest.GetUrl().SetPath(request.GetUrl().GetPath());
+
+    // Copy all headers
+    for (auto& header : request.GetHeaders())
+    {
+      redirectRequest.SetHeader(header.first, header.second);
+    }
+    // QP
+    for (auto const& qp : request.GetUrl().GetQueryParameters())
+    {
+      redirectRequest.GetUrl().AppendQueryParameter(qp.first, qp.second);
+    }
+    // Set x-recording-upstream-base-uri
+    {
+      auto const& url = request.GetUrl();
+      auto const port = url.GetPort();
+      auto const host
+          = url.GetScheme() + "://" + url.GetHost() + (port != 0 ? ":" + std::to_string(port) : "");
+      redirectRequest.SetHeader("x-recording-upstream-base-uri", host);
```

Comment:
Isn't this the same as `GetUrlWithoutQuery()`, or `GetAbsoluteUrl()`?

Reply:
GetUrlWithoutQuery() ->  includes the url path

GetAbsoluteUrl() -> includes path and query